configuration_test.go

```diff
@@ -57,6 +57,30 @@ func TestConfigurationTopic(t *testing.T) {
 	}
 }
 
+func TestConfigurationTopicWithPrefix(t *testing.T) {
+	cases := []struct {
```

Comment:
Just to be sure: this will still work without the prefix as well, right?

Reply:
Yes, without the prefix option it works just the way it was working before, there's a testcase before this one that checks it.

But if you want, we can get rid of topic_prefix in favour of a more structured topic structure that comes by default, that will probably make it easier to introduce mqtt discovery in the long run. So do let me know if you want me to transform topic_prefix into a new topic formatting scheme that you suggested in one of your previous comments.

Let me know if you want me to split the series into smaller PRs so that it becomes easier to merge. I think there are a few patches that are trivial and could go in immediately whereas there's a small number that would take more time to mature.